### PR TITLE
Register module

### LINKED ISSUE #3: Allow config of hm_platform global from Platform config
Once we have module registration in https://github.com/humanmade/platform-dev/issues/292 we should consider creating module settings for all the hm_platform globals.

### PR DESCRIPTION
**Pull request by joehoyle**
Make use of the module registration to handle the Cloud config. We still support `global $hm_platform` for backwards compat for now. I also switched X-Ray to be loaded from composer, which we'll want to do for the other deps at some point.

Fixes #3

### CODE REVIEW

File: bin/install-deps.sh [hunk 1]
```diff
@@ -1,6 +1,5 @@
 curl -L https://github.com/humanmade/Cavalcade/archive/907c2ec3e52bffe50a8d06ce589d1376731f1e2d.zip -o cavalcade.zip
 curl -L https://github.com/humanmade/aws-ses-wp-mail/archive/ae12da33c381bf04ce31c1b520e7b62c9d2269a7.zip -o aws-ses-wp-mail.zip
-curl -L https://github.com/humanmade/aws-xray/archive/b647c35a54fa59e3753758507e9415521ad7008d.zip -o aws-xray.zip
 curl -L https://github.com/humanmade/S3-Uploads/archive/237e73fef2cb56630afa9bb561fc121e58eb4886.zip -o s3-uploads.zip
 curl -L https://github.com/pantheon-systems/wp-redis/archive/ac5f4e62a298ea784f682c3ea5dc8061aedf525e.zip -o wp-redis.zip
```
[the remark this answers is not in the document]
Comment (joehoyle) [reply bin/install-deps.sh]: wp-redis is different to `wp-redis-predis-client`

**Comment by joehoyle** [comment]
@roborourke we already have "solved" the shared AWS SDK / Guzzle, we don't ship those are part of S3-Uploads, we use the autoloaded AWS SDK and Guzzle that is a "real" composer dep.

I created https://github.com/humanmade/platform-cloud/issues/5 to cover migrating some of those plugins over to composer deps, but that's separate to this PR (all those plugins are already added)

File: composer.json [hunk 2]
```diff
@@ -16,6 +16,7 @@
 		]
 	},
 	"require": {
-		"humanmade/wp-redis-predis-client": "dev-master"
+		"humanmade/wp-redis-predis-client": "dev-master",
+		"humanmade/aws-xray": "1.1.2"
```
[the remark this answers is not in the document]
Comment (joehoyle) [reply composer.json]: Created https://github.com/humanmade/platform-cloud/issues/5 to track

**Comment by roborourke** [comment]
Good to go 👍

**Pull request merged**